File: agent/sdk/typescript/examples/payment-with-verification.ts
import {
  Client,
  VerificationLevel,
  requireVerified,
  type IdempotentMutation,
  type SubmissionState,
  type VerifiedRead,
} from "../src/index.js";

export interface PaymentRequest { readonly canonicalBytes: Uint8Array }

export async function paymentWithVerification(
  client: Client,
  request: IdempotentMutation<PaymentRequest>,
): Promise<VerifiedRead<SubmissionState>> {
  const result = await client.call<typeof request, VerifiedRead<SubmissionState>>("submit", request);
  const verified = requireVerified(VerificationLevel.SequencerSigned, result);
  if (verified.value.kind !== "Executed") throw new Error(`payment_not_executed:${verified.value.kind}`);
  return verified;
}

File: agent/sdk/python/examples/payment_with_verification.py
from layerx_sdk import (
    Client,
    IdempotentMutation,
    SubmissionExecuted,
    SubmissionState,
    VerificationLevel,
    VerifiedRead,
    require_verified,
)


def payment_with_verification(
    client: Client,
    request: IdempotentMutation[bytes],
) -> VerifiedRead[SubmissionState]:
    result = client.call("submit", request)
    if not isinstance(result, VerifiedRead):
        raise TypeError("invalid_submit_response")
    verified = require_verified(VerificationLevel.SEQUENCER_SIGNED, result)
    if not isinstance(verified.value, SubmissionExecuted):
        raise RuntimeError(f"payment_not_executed:{verified.value.kind}")
    return verified
